File: src/foam/core/benchmark/PMBenchmark.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.core.benchmark;

import foam.lang.X;
import foam.core.bench.Benchmark;
import foam.core.bench.BenchmarkRunner;
import foam.core.bench.BenchmarkRunner.Builder;
import foam.core.pm.*;
import java.util.Map;

public class PMBenchmark
  extends Benchmark
{
  @Override
  public void execute(X x) {
    PM pm = new PM(foam.core.bench.Benchmark.class, "abc");
    pm.log(x);
  }
}

File: src/foam/core/benchmark/PMBenchmark.js
/**
 * @license
 * Copyright 2025 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.core.benchmark',
  name: 'PMBenchmark',
  extends: 'foam.core.bench.Benchmark',

  javaImports: [
    'foam.lang.X',
    'foam.core.bench.Benchmark',
    'foam.core.bench.BenchmarkRunner',
    'foam.core.bench.BenchmarkRunner.Builder',
    'foam.core.pm.*',
    'java.util.Map'
  ],

  methods: [
    {
      name: 'execute',
      args: [
        {
          name: 'x',
          type: 'X'
        }
      ],
      javaCode: `
        PM pm = new PM(foam.core.bench.Benchmark.class, "abc");
        pm.log(x);
      `
    }
  ]
});